src/main/java/com/coveo/saml/SamlClient.java

```diff
@@ -427,6 +432,49 @@ public static SamlClient fromMetadata(
         samlBinding);
   }
 
+  /**
+   * Wrap a {@link java.io.Reader Reader} to skip a BOM if it is present.
+   * OpenSaml won't accept a metadata file if it starts with a BOM.
+   * @param metadata The metadata with optional BOM
+   * @return A {@link BufferedReader} set to the first real character
+  */
+  private static BufferedReader skipBom(Reader metadata) throws SamlException {
+    BufferedReader bufferedMetadata = new BufferedReader(metadata);
+    try {
+      String bom = "\uFEFF";
```

Comment:
This only covers one type of BOM, those can have different lengths depending on the encoding.

Maybe have a look at https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/input/BOMInputStream.html ?

Reply:
The reason I didn't want to go with BOMInputStream was that I was worried all the wrapping of Readers and InputStreams might become unreadable. I figured the best way to show this would be to write the code. The resulting code is much more readable, so I'll commit that.

I do want to point out that the manual method does work for all Unicode BOMs. The Reader we wrap around metadata takes an encoding parameter, which ensures the BOM is decoded to the Unicode BOM code point. When we read one character we get the entire BOM character, which is the same BOM regardless of the source encoding.